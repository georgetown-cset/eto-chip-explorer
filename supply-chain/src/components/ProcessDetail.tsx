import React from "react";
import ReactMarkdown from 'react-markdown'
import mdxComponents from "../helpers/mdx_style";
import ProviderTable from "./ProviderTable";
import tooltips from "../helpers/tooltips";

export interface ProcessDetailProps {
  description: string;
  orgs: Record<string, number|string>;
  orgMeta: Record<string, {
    name: string;
    type: string;
    hq_flag: string;
    hq_country: string;
  }>;
}

const ProcessDetail = ({
  description,
  orgs = {},
  orgMeta,
}: ProcessDetailProps) => {
  const orgList = Object.keys(orgs).map(org => ({
    provider: orgMeta[org].name,
    flag: orgMeta[org].hq_flag,
  }));

  return (
    <div
      style={{display: "inline-block", padding: "0px 40px", textAlign: "left"}}
      className="process-detail"
    >
      <ReactMarkdown components={mdxComponents}>{description}</ReactMarkdown>
      <ProviderTable
        caption="Notable supplier companies"
        providers={orgList}
        tooltip={tooltips.providers.orgs}
      />
    </div>
  )
};

export default ProcessDetail;
