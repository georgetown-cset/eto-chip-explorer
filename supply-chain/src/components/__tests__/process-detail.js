import React from "react"
import renderer from "react-test-renderer"

import ProcessDetail from "../process_detail"

const descriptions = [
  {"slug": "N59", "body": "Some text about N59"},
  {"slug": "N35", "body": "Some text about N35"},
]

describe("Process detail", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<ProcessDetail selectedNode="N35" descriptions={descriptions}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
