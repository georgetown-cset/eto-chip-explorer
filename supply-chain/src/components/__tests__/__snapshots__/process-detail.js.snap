// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process detail renders correctly 1`] = `
<DocumentFragment>
  <div
    class="process-detail"
    style="display: inline-block; padding: 0px 40px; text-align: left;"
  >
    <div>
      Some text about N35
    </div>
    <div />
  </div>
</DocumentFragment>
`;
